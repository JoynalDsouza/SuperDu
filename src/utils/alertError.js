import {Alert} from 'react-native';

export const alertError = error => {
  try {
    const title = error?.title || 'Error';
    Alert.alert(title, error.message || error);
  } catch (e) {}
};
