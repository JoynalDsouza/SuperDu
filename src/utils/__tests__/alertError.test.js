import {alertError} from '../alertError';
import {Alert} from 'react-native';

// Path: src/utils/__tests__/alertError.test.js

describe('alertError function testing', () => {
  it('should match the snapshot', () => {
    expect(alertError).toMatchSnapshot();
  });
  it('should call Alert.alert with "Error" and error message', () => {
    const error = new Error('Test error message');
    const alertSpy = jest.spyOn(Alert, 'alert');
    alertError(error);

    expect(alertSpy).toHaveBeenCalledWith('Error', error.message);
  });
  it('should call Alert.alert with "Error" and error message', () => {
    const error = 'Test error message';
    const alertSpy = jest.spyOn(Alert, 'alert');
    alertError(error);
    expect(alertSpy).toHaveBeenCalledWith('Error', error);
  });
  it('if no error parameter is passed, it should not call anything', () => {
    const res = alertError();
    expect(res).toBeUndefined();
  });
});
