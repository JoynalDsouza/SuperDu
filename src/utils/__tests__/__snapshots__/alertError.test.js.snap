// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alertError function testing should match the snapshot 1`] = `[Function]`;
